qa/rolling-upgrade/src/test/java/org/opensearch/knn/bwc/PainlessScriptScoringIT.java

```diff
@@ -0,0 +1,116 @@
+                addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, docId, NUM_DOCS);
+                break;
+            case MIXED:
+                if (isFirstMixedRound()) {
+                    queryCount = NUM_DOCS;
+                    String sourceFirstMixedRound = createL2PainlessScriptSource(TEST_FIELD, DIMENSIONS, queryCount);
+                    validateKNNPainlessScriptScoreSearch(testIndex, TEST_FIELD, sourceFirstMixedRound, queryCount, K);
+
+                    docId = NUM_DOCS;
+                    addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, docId, NUM_DOCS);
+
+                    queryCount = 2 * NUM_DOCS;
+                    String updatedSourceFirstMixedRound = createL2PainlessScriptSource(TEST_FIELD, DIMENSIONS, queryCount);
+                    validateKNNPainlessScriptScoreSearch(testIndex, TEST_FIELD, updatedSourceFirstMixedRound, queryCount, K);
+                } else {
+                    queryCount = 2 * NUM_DOCS;
+                    String sourceSecondMixedRound = createL2PainlessScriptSource(TEST_FIELD, DIMENSIONS, queryCount);
+                    validateKNNPainlessScriptScoreSearch(testIndex, TEST_FIELD, sourceSecondMixedRound, queryCount, K);
+
+                    docId = 2 * NUM_DOCS;
+                    addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, docId, NUM_DOCS);
+
+                    queryCount = 3 * NUM_DOCS;
+                    String updatedSourceSecondMixedRound = createL2PainlessScriptSource(TEST_FIELD, DIMENSIONS, queryCount);
+                    validateKNNPainlessScriptScoreSearch(testIndex, TEST_FIELD, updatedSourceSecondMixedRound, queryCount, K);
+                }
+                break;
+            case UPGRADED:
+                queryCount = 3 * NUM_DOCS;
+                String sourceUpgraded = createL2PainlessScriptSource(TEST_FIELD, DIMENSIONS, queryCount);
+                validateKNNPainlessScriptScoreSearch(testIndex, TEST_FIELD, sourceUpgraded, queryCount, K);
+
+                docId = 3 * NUM_DOCS;
+                addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, docId, NUM_DOCS);
+
+                queryCount = 4 * NUM_DOCS;
+                String updatedSourceUpgraded = createL2PainlessScriptSource(TEST_FIELD, DIMENSIONS, queryCount);
+                validateKNNPainlessScriptScoreSearch(testIndex, TEST_FIELD, updatedSourceUpgraded, queryCount, K);
+
+                deleteKNNIndex(testIndex);
+        }
+    }
+
+    // KNN painless script scoring for space_type "l1"
+    public void testKNNL1PainlessScriptScore() throws Exception {
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+        switch (getClusterType()) {
+            case OLD:
+                createKnnIndex(testIndex, createKNNDefaultScriptScoreSettings(), createKnnIndexMapping(TEST_FIELD, DIMENSIONS));
+                addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, docId, NUM_DOCS);
+                break;
+            case MIXED:
+                if (isFirstMixedRound()) {
```

Comment:
Seems like we could avoid a if/else block, if we just set a variable based on what round it is.

Reply:
Could you please explain?

Reply:
In the mixed case, a lot of the time, code appears to be duplicated:
```

if (isFirstMixedRound()) {
                    totalDocsCount = NUM_DOCS;
                    String sourceFirstMixedRound = createL2PainlessScriptSource(TEST_FIELD, DIMENSIONS, totalDocsCount);
                    validateKNNPainlessScriptScoreSearch(testIndex, TEST_FIELD, sourceFirstMixedRound, totalDocsCount, K);

                    docId = NUM_DOCS;
                    addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, docId, NUM_DOCS);

                    totalDocsCount = 2 * NUM_DOCS;
                    String updatedSourceFirstMixedRound = createL2PainlessScriptSource(TEST_FIELD, DIMENSIONS, totalDocsCount);
                    validateKNNPainlessScriptScoreSearch(testIndex, TEST_FIELD, updatedSourceFirstMixedRound, totalDocsCount, K);
} else {
                    totalDocsCount = 2 * NUM_DOCS;
                    String sourceSecondMixedRound = createL2PainlessScriptSource(TEST_FIELD, DIMENSIONS, totalDocsCount);
                    validateKNNPainlessScriptScoreSearch(testIndex, TEST_FIELD, sourceSecondMixedRound, totalDocsCount, K);

                    docId = 2 * NUM_DOCS;
                    addKNNDocs(testIndex, TEST_FIELD, DIMENSIONS, docId, NUM_DOCS);

                    totalDocsCount = 3 * NUM_DOCS;
                    String updatedSourceSecondMixedRound = createL2PainlessScriptSource(TEST_FIELD, DIMENSIONS, totalDocsCount);
                    validateKNNPainlessScriptScoreSearch(testIndex, TEST_FIELD, updatedSourceSecondMixedRound, totalDocsCount, K);
}
```

I see this in the tests a couple of times. Instead of writing most of the code twice, set variables that change with the round and use those variables in the common logic.

Reply:
Refer to comment above.